assembly/src/ast/imports.rs

```diff
@@ -51,13 +51,12 @@ impl ModuleImports {
         while let Some(token) = tokens.read() {
             match token.parts()[0] {
                 Token::USE => {
-                    let module_path = token.parse_use()?;
-                    let module_name = module_path.last();
-                    if imports.contains_key(module_name) {
+                    let (module_path, module_name) = token.parse_use()?;
+                    if imports.contains_key(&module_name) {
                         return Err(ParsingError::duplicate_module_import(token, &module_path));
                     }
 
-                    imports.insert(module_name.to_string(), module_path);
+                    imports.insert(module_name, module_path);
```

Comment:
Question: what would happen if we import the same module directly and under an alias (or under two different aliases)? Maybe we should explicitly prohibit this case as I don't see any reason for why someone would want to do it.

Reply:
Thanks, I missed that. I think we'll need to check against module paths now which is a little inefficient but should probably be acceptable in this case.

Reply:
Yep - I think this is fine this case.